include/pulsar/ConsumerConfiguration.h

```diff
@@ -383,6 +383,26 @@ class PULSAR_PUBLIC ConsumerConfiguration {
      */
     InitialPosition getSubscriptionInitialPosition() const;
 
+    /**
+     * Selects the subscription mode to be used when subscribing to a topic.
+     *
+     * <p>Options are:
+     * <ul>
+     *  <li>{@link SubscriptionMode#Durable} (Default)</li>
+     *  <li>{@link SubscriptionMode#NonDurable}</li>
+     * </ul>
+     *
+     * @param subscriptionMode the subscription mode value
+     */
+    void setSubscriptionMode(SubscriptionMode subscriptionMode);
```

Comment:
```suggestion
    ConsumerConfiguration& setSubscriptionMode(SubscriptionMode subscriptionMode);
```

Support method chaining.

Reply:
I noticed that the style was different. Some returned config&, and some did not. I will unify it.